ledger/sandbox/src/test/lib/scala/com/digitalasset/platform/store/dao/JdbcLedgerDaoTransactionTreesSpec.scala

```diff
@@ -0,0 +1,158 @@
+          transaction.offset shouldBe ApiOffset.toApiString(offset)
+          transaction.transactionId shouldBe tx.transactionId
+          transaction.workflowId shouldBe tx.workflowId.getOrElse("")
+          val created = transaction.eventsById.values.loneElement.getCreated
+          transaction.rootEventIds.loneElement shouldEqual created.eventId
+          created.eventId shouldBe eventId
+          created.witnessParties should contain only tx.submittingParty.get
+          created.agreementText.getOrElse("") shouldBe createNode.coinst.agreementText
+          created.contractKey shouldBe None
+          created.createArguments shouldNot be(None)
+          created.signatories should contain theSameElementsAs createNode.signatories
+          created.observers should contain theSameElementsAs createNode.stakeholders.diff(
+            createNode.signatories)
+          created.templateId shouldNot be(None)
+      }
+    }
+  }
+
+  it should "return the expected flat transaction for a correct request (exercise)" in {
+    for {
+      (_, create) <- storeCreateTransaction()
+      created <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(create.transactionId, Set(create.submittingParty.get))
+      (offset, exercise) <- storeExerciseTransaction(
+        AbsoluteContractId(created.get.transaction.get.eventsById.head._2.getCreated.contractId))
+      result <- ledgerDao.transactionsReader
+        .lookupTransactionTreeById(exercise.transactionId, Set(exercise.submittingParty.get))
+    } yield {
+      inside(result.value.transaction) {
+        case Some(transaction) =>
+          val (eventId, exerciseNode: NodeExercises.WithTxValue[EventId, AbsoluteContractId]) =
+            exercise.transaction.nodes.head
+          transaction.commandId shouldBe exercise.commandId.get
+          transaction.offset shouldBe ApiOffset.toApiString(offset)
+          transaction.transactionId shouldBe exercise.transactionId
+          transaction.workflowId shouldBe exercise.workflowId.getOrElse("")
+          val exercised = transaction.eventsById.values.loneElement.getExercised
+          transaction.rootEventIds.loneElement shouldEqual exercised.eventId
+          exercised.eventId shouldBe eventId
+          exercised.witnessParties should contain only exercise.submittingParty.get
+          exercised.contractId shouldBe exerciseNode.targetCoid.coid
+          exercised.templateId shouldNot be(None)
+          exercised.actingParties should contain theSameElementsAs exerciseNode.actingParties
+          exercised.childEventIds shouldBe Seq.empty
+          exercised.choice shouldBe exerciseNode.choiceId
+          exercised.choiceArgument shouldNot be(None)
+          exercised.consuming shouldBe true
+          exercised.exerciseResult shouldNot be(None)
+      }
+    }
+  }
+
+  it should "return the expected flat transaction for a correct request (create, exercise)" in {
```

Comment:
This test reads the transaction tree, not the flat transaction.

Reply:
Yes, thanks, copy-pasta.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/5089/commits/eba57bf61d9a7587aa53a066fc83b479c6980983